fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1505,19 +1506,17 @@ private FHIRPersistenceNotSupportedException buildNotSupportedException(String m
             SingleResourceResult<T> result = new SingleResourceResult.Builder<T>()
                     .success(true)
                     .resource(resource)
+                    .deleted(resourceIsDeleted)
                     .build();
 
             return result;
-        }
-        catch(FHIRPersistenceResourceDeletedException e) {
+        } catch(FHIRPersistenceResourceDeletedException e) {
```

Comment:
I wonder if it would be possible to remove this exception now that we have the idDeleted marker in the response object...

Reply:
We still need to throw it: 
```
            if (resourceIsDeleted && !context.includeDeleted()) {
                throw new FHIRPersistenceResourceDeletedException("Resource '" +
                        resourceType.getSimpleName() + "/" + logicalId + "' is deleted.");
            }
```

Reply:
OK.  I'm definitely fine with this...I've just been looking out for chances to remove exception-handling from the normal request processing path where possible.